api/src/main/kotlin/com/google/devtools/ksp/processing/CodeGenerator.kt

```diff
@@ -56,16 +72,28 @@ interface CodeGenerator {
      * @see [CodeGenerator] for more details.
      */
     fun associate(sources: List<KSFile>, packageName: String, fileName: String, extensionName: String = "kt")
+}
+
+/**
+ * Dependencies of an output file.
+ */
+class Dependencies private constructor(val isAllSources: Boolean, val dependOnNewChanges: Boolean, val sources: List<KSFile>) {
```

Comment:
OriginatingFiles might be a better name: https://docs.oracle.com/javase/7/docs/api/javax/annotation/processing/Filer.html#createClassFile(java.lang.CharSequence,%20javax.lang.model.element.Element...)

or originating nodes if the API received nodes instead of files.

Reply:
Done.